pontoon/base/models.py

```diff
@@ -377,6 +357,87 @@ def add_translation(self, string, locale, user):
 
         translation.save()
 
+    def get_translation(self, plural_form=None):
+        """Get fetched translation of a given entity."""
+        translations = self.fetched_translations
+
+        if translations:
+            if plural_form:
+                translations = [t for t in translations if t.plural_form == plural_form]
+
+            translation = sorted(translations, key=lambda k: (-k.approved, k.date))[0]
```

Comment:
If there are no translations matching the requested plural form, this will throw an `IndexError`. You should probably do the plural filtering before the `if translations` check.